src/components/Fund/FundForm/FundForm.js

```diff
@@ -0,0 +1,321 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  Field,
+  getFormValues,
+} from 'redux-form';
+import {
+  find,
+  get,
+} from 'lodash';
+
+import { stripesShape } from '@folio/stripes/core';
+import stripesForm from '@folio/stripes/form';
+import {
+  Accordion,
+  AccordionSet,
+  Button,
+  Col,
+  ExpandAllButton,
+  Pane,
+  PaneMenu,
+  Paneset,
+  Row,
+  TextArea,
+  TextField,
+} from '@folio/stripes/components';
+import { ViewMetaData } from '@folio/stripes/smart-components';
+import {
+  FieldMultiSelection,
+  FieldSelection,
+  useAccordionToggle,
+  validateRequired,
+} from '@folio/stripes-acq-components';
+
+import {
+  FUND_FORM,
+  SECTIONS_FUND,
+  FUND_STATUSES_OPTIONS,
+} from '../constants';
+
+const getLastMenu = (handleSubmit, pristine, submitting) => {
+  return (
+    <PaneMenu>
+      <Button
+        buttonStyle="primary"
+        data-test-button-save-fund
+        disabled={pristine || submitting}
+        marginBottom0
+        onClick={handleSubmit}
+        type="submit"
+      >
+        <FormattedMessage id="ui-finance.fund.save" />
+      </Button>
+    </PaneMenu>
+  );
+};
+const itemToString = item => item;
+
+const FundForm = ({
+  change,
+  dispatch,
+  handleSubmit,
+  initialValues,
+  onCancel,
+  parentResources,
+  pristine,
+  stripes,
+  submitting,
+}) => {
+  const [expandAll, sections, toggleSection] = useAccordionToggle();
+
+  const selectLedger = useCallback(
+    (e, selectedLedgerId) => (
+      dispatch(change('ledgerId', selectedLedgerId))
+    ),
+    [],
+  );
+
+  const funds = get(parentResources, ['records', 'records'], []);
+  const fundTypes = get(parentResources, ['fundTypes', 'records'], []).map(
+    ({ name, id }) => ({
+      label: name,
+      value: id
+    })
+  );
+  const ledgers = get(parentResources, ['ledgers', 'records'], []).map(
+    ({ name, id, currency }) => ({
+      label: name,
+      value: id,
+      currency,
+    })
+  );
+  const lastMenu = getLastMenu(handleSubmit, pristine, submitting);
+  const paneTitle = initialValues.id
+    ? initialValues.name
+    : <FormattedMessage id="ui-finance.fund.paneTitle.create" />;
+  const metadata = initialValues.metadata;
+  const formValues = getFormValues(FUND_FORM)(stripes.store.getState());
+  const selectedLedger = find(ledgers, ['value', formValues.ledgerId]);
+  const fundCurrency = get(selectedLedger, 'currency', '');
+  const fundOptions = funds.map(({ id }) => id);
+
+  const formatter = ({ option }) => {
+    const item = find(funds, { id: option }) || option;
+
+    if (!item) return option;
+
+    return item.name;
+  };
+
+  const filter = (filterText, list) => {
+    const renderedItems = filterText
+      ? funds
```

Comment:
should it be `funds`, or you can filter passed in `list`? I'm not familiar what comes as a `list` though

Reply:
Yes. `list` it's an array of fund ids only